pkg/hash/hash_test.go

```diff
@@ -0,0 +1,80 @@
+package hash_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-psap/special-resource-operator/pkg/hash"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const emptyHash = "12161962213042174405"
+
+func TestHash(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Hash Suite")
+}
+
+var _ = Describe("Hash", func() {
+	Context("TestFNV64a", func() {
+		When("empty string", func() {
+			s, err := hash.FNV64a("")
+
+			It("should work as expected", func() {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(s).To(Equal("cbf29ce484222325"))
+			})
+		})
+
+		When("random string", func() {
+			s, err := hash.FNV64a("special-resource-operator")
+
+			It("should work as expected", func() {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(s).To(Equal("20db61ac8744a54a"))
+			})
+		})
+	})
+
+	Context("Annotate", func() {
+		It("should work as expected", func() {
+			obj := &unstructured.Unstructured{}
+
+			err := hash.Annotate(obj)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(obj.GetAnnotations()["specialresource.openshift.io/hash"]).To(Equal(emptyHash))
+		})
+	})
+
+	Context("AnnotationEqual", func() {
+		var objOld, objNew *unstructured.Unstructured
+
+		BeforeEach(func() {
+			objOld = &unstructured.Unstructured{}
+			objNew = &unstructured.Unstructured{}
+		})
+
+		When("bad annotation", func() {
```

Comment:
i don't think you need a "When" here

Reply:
So something like:
```go
 		It("should return false with a bad annotation", func() {
```
Is that what you would do?

Reply:
Fixed with a table test.